source/extensions/upstreams/tcp/generic/config.cc

```diff
@@ -16,7 +17,14 @@ TcpProxy::GenericConnPoolPtr GenericConnPoolFactory::createGenericConnPool(
     TcpProxy::TunnelingConfigHelperOptConstRef config, Upstream::LoadBalancerContext* context,
     Envoy::Tcp::ConnectionPool::UpstreamCallbacks& upstream_callbacks,
     StreamInfo::StreamInfo& downstream_info) const {
-  if (config.has_value()) {
+  const StreamInfo::BooleanAccessor* disable_tunneling =
+      downstream_info.filterState()->getDataReadOnly<StreamInfo::BooleanAccessor>(
+          "envoy.tcp_proxy.disable_tunneling");
```

Comment:
Need a constant defined I think

Reply:
done

Reply:
@kyessenov I added a constant, but in tcp_proxy.h. When I come to think about it, the tcp_proxy implementation is receiving a generic upstream from the generic pool factory (which is an extension, in extensions/upstreams/tcp/generic/config.cc) and the logic to inspect the existence of that filter state key, is at the factory implementation. Do you think I should move the constant definition there and rename?

On second hand, if it's defined in the extensions, generally it won't be accessible to external consumers due to visibility, although in this particular case the "config" target is defined as public.

 WDYT? Please help me decide on this.

Reply:
I think it should be defined at the point of use, which is tcp_proxy in common code. That way it's shared and accessible everywhere.